File: core/src/nl/t64/game/rpg/subjects/MapObserver.java
package nl.t64.game.rpg.subjects;

import com.badlogic.gdx.graphics.Color;
import nl.t64.game.rpg.screens.world.GameMap;


public interface MapObserver {

    void onNotifyMapWillChange(Runnable changeMap, Color transitionColor);

    void onNotifyMapChanged(GameMap currentMap);

    void onNotifyShakeCamera();

}

File: core/src/nl/t64/game/rpg/subjects/MapObserver.kt
package nl.t64.game.rpg.subjects

import com.badlogic.gdx.graphics.Color
import nl.t64.game.rpg.screens.world.GameMap


interface MapObserver {

    fun onNotifyFadeOut(actionAfterFade: () -> Unit, transitionColor: Color)
    fun onNotifyMapChanged(currentMap: GameMap)
    fun onNotifyShakeCamera()
    fun onNotifyStartCutscene(cutsceneId: String)

}
